contracts/peripheral/YieldDirector.sol

```diff
@@ -427,17 +443,18 @@ contract YieldDirector is IYieldDirector, YieldSplitter, OlympusAccessControlled
     /**
         @notice Redeem available gOHM yield from a specific deposit
         @param depositId_ Deposit ID to withdraw gOHM yield from
+        @param recipient_ address of recipient
     */
-    function _redeem(uint256 depositId_) internal returns (uint256 amountRedeemed) {
+    function _redeem(uint256 depositId_, address recipient_) internal returns (uint256 amountRedeemed) {
         if (redeemDisabled) revert YieldDirector_RedeemsDisabled();
-        if (recipientLookup[depositId_] != msg.sender) revert YieldDirector_NotYourYield();
+        if (recipientLookup[depositId_] != recipient_) revert YieldDirector_NotYourYield();
```

Comment:
I think this tracks as still being protected from someone else redeeming your yield, was it thoroughly tested though?

Reply:
So i changed those internal functions to use parameter recipient because we now have a function that allows external parties to redeem on behalf of users. we want to use passed in recipient rather than msg.sender. In most cases redeemer is msg.sender, but it will be passed in.